src/Scores.cpp

```diff
@@ -154,6 +160,91 @@ void ScoreHolder::printPeptide(ostream& os, bool printDecoys, bool printExpMass,
+  /*  uncalibrated_precursor_neutral_mass ? */
+  double expMass = pPSM->expMass;
+  /*  precursor_neutral_mass ? */
+  double calcMass = pPSM->calcMass;
+
+  os << "    <spectrum_query spectrum=\"" << id << "\" precursor_neutral_mass=\"" << expMass << "\" assumed_charge=\"" << assumed_charge << "\" end_scan=\"" << scan << "\" index=\"" << index << "\" retention_time_sec=\"" << fixed << setprecision (3) << pPSM->getRetentionTime() << "\" start_scan=\"" << scan << "\">" << endl;
+  std::string centpep = pPSM->getPeptideSequence();
+  std::string trimmed_pep = trim_left_copy_if(centpep, is_any_of("n"));
+  regex r("\\[(.*?)\\]");
+  std::string peptide_sequence = regex_replace(trimmed_pep, r, "");
+  os << "    <search_result>" << endl;
+
+  /* Print protein information */
+  size_t n_protein = 0;
+  /* Placeholders */
+  int hit_rank = 1;
+  int massdiff = 1;
+  /* num_tot_proteins */
+  int num_tot_proteins = pPSM->proteinIds.size();
+
+  std::vector<std::string>::const_iterator pidIt = pPSM->proteinIds.begin();
+  for ( ; pidIt != pPSM->proteinIds.end() ; ++pidIt) {
+    if (n_protein==0) {
+      /*  set calc_neutral_pep_mass  as calcMass as placeholder for now */
+      os << "    <search_hit calc_neutral_pep_mass=\"" << calcMass << "\" num_tot_proteins=\"" << num_tot_proteins << "\" hit_rank=\""<< hit_rank <<"\" massdiff=\"" << massdiff << "\" peptide=\"" << peptide_sequence << "\" protein=\"" << getRidOfUnprintablesAndUnicode(*pidIt) << "\">" << endl;
+    } else {
+      os << "    <alternative_protein protein=\"" << getRidOfUnprintablesAndUnicode(*pidIt) << "\"/>" << endl;
+    }
+    n_protein++;
+  }
+  /* Print modification */
+  string subject(trimmed_pep);
+  smatch match;
+  long int mod_pos = 0;
+  float mod_weight;
+  size_t n_mod = 0;
+
+  while (regex_search(subject, match, r)) {
+    mod_pos += match.position(0);
+    // suffix to find the rest of the string.
+    if (mod_pos==0) {
+      mod_weight = std::stof(match.str(1)) + 1.0074;
+      if (n_mod ==0) {
+        os << "    <modification_info mod_nterm_mass=\"" << round(mod_weight * 1000) / 1000 <<"\">" << endl;
+      }
+    } else {
+      mod_weight = aaWeight[peptide_sequence.at(mod_pos - 1)] + std::stof(match.str(1));
+      if (n_mod == 0) {
+        os << "    <modification_info>" << endl;
+      }
+      os << "    <mod_aminoacid_mass mass=\""<< round(mod_weight * 1000) / 1000  <<"\" position=\""<< std::to_string(mod_pos) << "\" />" << endl;
+    }
+    subject = match.suffix().str();
+    n_mod++;
+  }
+  if (n_mod != 0) {
+    os << "    </modification_info>" << endl;
+  }
+
+  /* Print Percolator information */ 
+  os << "    <analysis_result analysis=\"peptideprophet\">" << endl;
+  os << "    <peptideprophet_result probability=\"" << scientific << 1.0 - pep   << "\" />" << endl;
+  os << "    </analysis_result>" << endl; 
+  os << "    </search_hit>"<< endl;
+  os << "    </search_result>" << endl;
+  os << "    </spectrum_query>" << endl;
```

Comment:
Try to make the indentations in the XML file such that it is easily readable what the parent-child relations are, e.g.
```
<search_hit>
  <analysis_result>
    <peptideprophet_result />
  </analysis_result>
</search_hit>
``` 

This also applies to all other places where you're writing XML.

Reply:
Fixed